GoogleApiWrappers/src/Google.Apis.Storage.v1.ClientWrapper/DownloadObjectOptions.cs

```diff
@@ -25,5 +31,26 @@ internal void ModifyDownloader(MediaDownloader downloader)
                 downloader.ChunkSize = ChunkSize.Value;
             }
         }
+
+        /// <summary>
+        /// Appends any parameters to the given builder, which is guaranteed to
+        /// already have one query parameter (alt=media).
+        /// </summary>
+        internal void AppendUriParameters(StringBuilder uriBuilder)
+        {
+            MaybeAppendParameter(uriBuilder, "generation", Generation);
+        }
+
+        private static void MaybeAppendParameter(StringBuilder uriBuilder, string name, long? value)
+        {
+            if (value != null)
+            {
+                uriBuilder
+                    .Append("&")
```

Comment:
This assumes that it is legal to append the '&' character to the URI, that is that there's already a parameter in it. While it is true in the current code it might not always be in the general sense. Is it possible to check the contents of the `uriBuilder` to at least assert that this assumption holds?